back/src/Model/Space.ts

```diff
@@ -0,0 +1,90 @@
+import { CustomJsonReplacerInterface } from "./CustomJsonReplacerInterface";
+import { Pusher, SpaceMessage } from "./Pusher";
+import {
+    AddSpaceUserMessage,
+    RemoveSpaceUserMessage,
+    SpaceUser,
+    UpdateSpaceUserMessage,
+} from "../Messages/generated/messages_pb";
+import Debug from "debug";
+
+const debug = Debug("space");
+
+export class Space implements CustomJsonReplacerInterface {
+    readonly name: string;
+    private users: Map<Pusher, Map<string, SpaceUser>>;
```

Comment:
I'm not sure it makes sense to classify users by pusher.
You will almost never want to sort the users by pusher (unless when a pusher dies, but that is rare) and it will make your code more complex all over the place. For instance, when you want to send the complete list of users to a newly connected pusher.

If this is getting complexe, you can fallback to a simple `Map<string, SpaceUser>` and put the "Pusher" field inside the "SpaceUser" class.